superstring/tgreedy.py

```diff
@@ -50,6 +41,11 @@ def _mgreedy_first_phase(words):
       T.add(w_1)
   return T
 
+# achieves 4-approximation
+def mgreedy(words):
+  return next(iter(words), [''])[0] + \
```

Comment:
I have decided to make it a bit more complicated than the obvious solutios, so that it works even for words being empty. If I just took a wildcard from the first string it would then crash for words being empty. Also, I wanted to use the dummy character from a string inputted by the user as we do not have any prior knowledge about what kind of characters can be used as the dummy character.

Reply:
That's fine, as this is an educational library, not an industry one, we are allowed to assume certain conventions consistent with the lecture notes - but from a broader perspective I understand your cautiousness about the corner cases.